### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by dixyes** [dixyes/lwmbs#3] [General]
## why?

make micro and cli binaries easier to fetch and customize

## no issue?

lwmbs is not intended for end-user usage. If you are a distributor who wants to contribute to it, please file a pr, or share your ideas at the discussion of this repo.

## some extensions are not implemented

something is WIP, and some else won't be implemented (mostly they have licensed under GPL/LGPL/AGPL/other things that we cannot distribute statically)

the iconv extension will not be implemented in linux-glibc and windows binaries due to license (I'm too lazy to distribute all objects used in a micro build, that's too cumbersome)

## how to choose a sfx version

for linux, there are many variants of micro sfx (and cli), here is their meaning:

- static means all dependencies are statically linked into the sfx, which makes the sfx can run at bare kernel (e.g. docker "scratch" image), but this kind of micro sfx cannot use ffi and `dl()`
- shared means all dependencies except libc (musl or glibc) are statically linked into the sfx, which makes the sfx can do `dl()`/ffi, but you must prepare glibc or musl for the sfx to work
- glibc is the most common libc among the most popular linux distros, it's bigger and has better performance (in most scenes). since it's GPL-licensed, I cannot distribute it statically
- musl is a tiny and MIT-licensed libc, alpine and openwrt are using it, it's so small that I can statically link it into the sfx without a huge overhead
- x86_64 is the most common PC/server cpu architecture for now, if you have an Intel or AMD cpu, It's probably an x86_64 cpu
- aarch64 is the architecture name of arm64 cpus, it's the most common cpu type for SBC/phones, Apple's M1 M2 things are also aarch64

**Comment by codinglist** [comment]
for mac,acitons-workflow ，EXP.  we can see micro_8.2_x86_64_xxxx、 micro-cli_8.2_x86_64_xxxx、cli_8.2_x86_64_xxxx
what's the differences, what these means which those start with micro_ or micro-cli ?

**Reply by dixyes** [reply]
- micro is for [micro SAPI](https://github.com/dixyes/phpmicro)
- micro-cli is also micro SAPI but it's sapi name is "cli", this is for some libraries that checks sapi name
- cli is CLI SAPI, it's known as the "php" command

**Reply by codinglist** [reply]
Thanks
